syntax = "proto3";

// Mensagem para a identificação do equipamento
message EquipmentIdentification {
  enum EquipmentType {
    LAMP = 0;
    TV = 1;
    AIR_CONDITIONER = 2;
  }
  string equipment_id = 1;  // Um ID único para o equipamento
  EquipmentType type = 2;    // Tipo do equipamento (LAMP, TV, AIR_CONDITIONER)
  string ip_address = 3;     // Endereço IP do equipamento
  int32 port = 4;            // Porta na qual o equipamento está escutando
}

// Mensagem para o estado do equipamento (ligado/desligado)
message EquipmentState {
  string equipment_id = 1;  // ID único do equipamento
  bool is_on = 2;           // Indica se o equipamento está ligado (true) ou desligado (false)
}

// Mensagem para comandos de controle (ligar/desligar) enviados pelo Gateway aos equipamentos
message ControlCommand {
  string equipment_id = 1;  // ID único do equipamento
  bool turn_on = 2;         // Liga o equipamento se true, desliga se false
}

// Mensagem para dados de temperatura enviados pelo ar-condicionado ao Gateway
message TemperatureData {
  string equipment_id = 1;  // ID único do equipamento (ar-condicionado)
  float temperature = 2;     // Temperatura enviada pelo sensor do ar-condicionado
}

// Mensagem para a descoberta de equipamentos
message EquipmentDiscovery {
  string message = 1;  // Mensagem de descoberta enviada pelo Gateway
}
